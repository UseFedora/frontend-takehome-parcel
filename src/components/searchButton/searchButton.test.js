import React from 'react';
import { shallow } from 'enzyme';
import SearchButton from './';

describe('SearchButton', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchButton />);
    expect(component).toMatchSnapshot();
  });

  it('should trigger the click handler', () => {
    const spyOnClick = jest.fn();
    const component = shallow(
      <SearchButton
        onClick={spyOnClick}
      />
    );
    component.find('button').simulate('click');
    expect(spyOnClick).toHaveBeenCalled();
  });
});
