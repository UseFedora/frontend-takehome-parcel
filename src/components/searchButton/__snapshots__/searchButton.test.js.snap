// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchButton
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Search",
      "className": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Search",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Search",
        "className": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Search",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
